tools/applyconfig/applyconfig.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+type options struct {
+	confirm   bool
+	level     string
+	user      string
+	directory string
+}
+
+var adminConfig = regexp.MustCompile(`^admin_.+\.yaml$`)
+var standardConfig = regexp.MustCompile(`^cfg_.+\.yaml$`)
+
+func gatherOptions() *options {
+	opt := &options{}
+	flag.BoolVar(&opt.confirm, "confirm", false, "Set to true to make applyconfig commit the config to the cluster")
+	flag.StringVar(&opt.level, "level", "standard", "Select which config to apply (standard, admin, all)")
+	flag.StringVar(&opt.user, "as", "", "Username to impersonate while applying the config")
+	flag.StringVar(&opt.directory, "config-dir", "", "Directory with config to apply")
+
+	flag.Parse()
+
+	if opt.level != "standard" && opt.level != "admin" && opt.level != "all" {
+		fmt.Fprintf(os.Stderr, "--level: must be one of [standard, admin, all]\n")
+		os.Exit(1)
+	}
+
+	if opt.directory == "" {
+		fmt.Fprintf(os.Stderr, "--config-dir must be provided\n")
+		os.Exit(1)
+	}
+
+	return opt
+}
+
+func isAdminConfig(filename string) bool {
+	return adminConfig.MatchString(filename)
+}
+
+func isStandardConfig(filename string) bool {
+	return filepath.Ext(filename) == ".yaml" &&
```

Comment:
why do we do the ext check here but not for admin?
doesn't the regex enforce it?

Reply:
In admin the ext is enforced by the regex. "Standard" is any YAML that is not admin. I did it this way to make auto-apply opt-out rather that opt-in. This way we are even stricter: we attempt to apply everything unless you opt-out, making the opportunity to make a mistake and not apply something smaller. 

The JIRA has a _release repo has a presubmit that fails when someone adds config that is not applied by running the make targets_ item, and this is IMO a good way how to deal with that item (indirectly).